### DISCUSSION TITLE
Several inquiries

### DISCUSSION THREAD

**Discussion by keysongkang** [mzjb/DeepH-pack#44] [Q&A] [answered]
Hello, I have a few inquiries that I haven't been able to resolve on my own. If anyone has the answers or can provide any assistance, it would be greatly appreciated. Your help would be a great help to me!

1. I encountered a warning message during the preprocessing stage, and I'm unsure of its significance. It seems that there might be an empty entry in the Hamiltonian. I'm curious to know whether I should be concerned about this and if it requires any special attention. Any insights or advice would be greatly appreciated.
`u/kkang/.local/lib/python3.7/site-packages/deeph/rotate.py:203: UserWarning: Hamiltonian matrix block ([0, -1, 1, 3, 6]) do not have local coordinate
  warnings.warn(f'Hamiltonian matrix block ({key_str}) do not have local coordinate')`

2. I'm curious about the meaning of the orbital basis number in the "DeepH-pack/tools/get_all_orbital_str.py" script. Specifically, I would like to know if it represents an angular momentum number or if it has a different interpretation.

3. I have a question regarding the conversion of outputs from FHI-aims. Currently, it appears that the conversion with "DeepH-pack/deeph/preprocess/aims_get_data.jl" is done using the relative position parameter called "atom_frac," which is used to count the number of atoms. However, I'm wondering if there would be any issues if I were to change it to "atom" for an absolute Cartesian position instead. Would this modification potentially cause any problems or discrepancies in the output? 
 
I would appreciate any insights or advice on this matter. Thank you!

**Comment by mzjb** [comment, marked as answer]
@keysongkang 
Hi,

1. This warning indicates that you have truncated the Hamiltonian you want to predict, and some atomic pairs (ij) with relatively large interatomic distances do not need to be predicted in the corresponding Hamiltonian matrix block (Hij). This warning usually does not significantly affect the calculated properties of the final assembled Hamiltonian. Setting the radius to -1 in the configuration file of deeph-preprocess will disable truncation, and you can expect to avoid this warning.

2. DeepH uses a local coordinate system to transform rotationally equivariant Hamiltonian matrix blocks (Hij) into rotationally invariant blocks (Hij'). This allows the neural network to predict px/py/pz orbitals separately without considering the overall rotational form of the same angular momentum. Therefore, there is no need to assign orbital numbers based on angular momentum. The `orbital` parameter represents the numbering of each basis function. For example, if an element is described using s2p2d1, there are a total of 2x1 + 2x3 + 1x5 = 13 orbitals, and their numbers in `orbital` range from 0 to 12. In [DeepH-E3](https://www.nature.com/articles/s41467-023-38468-8), we do not perform the coordinate transformation to rotationally invariant Hij' using the local coordinate system. Instead, we directly predict Hij with a rotationally equivariant neural network. In this case, we need to predict px/py/pz orbitals as a whole, and the orbital numbers for the aforementioned example would range from 0 to 4.

3. I have not tried it, but I know that in another DFT program called OpenMX, choosing fractional or Cartesian coordinates can lead to different behaviors in the program, affecting whether the atoms are moved within the [0, 1) range.

**Reply by keysongkang** [reply]
Thank you so much!
